sgkit/display.py

```diff
@@ -0,0 +1,192 @@
+    else:
+        s_slice_0 = slice(0, max_samples // 2 + 1)
+        s_slice_1 = slice(n_sample - max_samples // 2, n_sample)
+
+    grid = [
+        [
+            arr.isel(variants=v_slice_0, samples=s_slice_0),
+            arr.isel(variants=v_slice_0, samples=s_slice_1),
+        ],
+        [
+            arr.isel(variants=v_slice_1, samples=s_slice_0),
+            arr.isel(variants=v_slice_1, samples=s_slice_1),
+        ],
+    ]
+    combined: xr.DataArray = xr.combine_nested(grid, concat_dim=["variants", "samples"])  # type: ignore[no-untyped-call]
+
+    variants_subset: xr.DataArray = xr.combine_nested([variants.isel(variants=v_slice_0), variants.isel(variants=v_slice_1)], concat_dim=["variants"])  # type: ignore[no-untyped-call]
+    samples_subset: xr.DataArray = xr.combine_nested([samples.isel(samples=s_slice_0), samples.isel(samples=s_slice_1)], concat_dim=["samples"])  # type: ignore[no-untyped-call]
+    combined = combined.assign_coords(variants=variants_subset)  # type: ignore[no-untyped-call]
+    combined = combined.assign_coords(samples=samples_subset)  # type: ignore[no-untyped-call]
+
+    assert combined.shape[0] <= max_variants + 1
+    assert combined.shape[1] <= max_samples + 1
+
+    return combined
+
+
+def display_genotypes(
+    ds: xr.Dataset, max_variants: int = 60, max_samples: int = 10
+) -> GenotypeDisplay:
+    """Display genotype calls.
+
+    Display genotype calls in a tabular format, with rows for variants,
+    and columns for samples. Genotypes are displayed in the same manner
+    as in VCF. For example, `1/0` is a diploid call of the first alternate
+    allele and the reference allele (0). Phased calls are denoted by a `|`
+    separator. Missing values are denoted by `.`.
+
+    Parameters
+    ----------
+    ds : Dataset
+        The dataset containing genotype calls in the `call/genotype`
+        variable, and (optionally) phasing information in the
+        `call/genotype_phased` variable. If no phasing information is
+        present genotypes are assumed to be unphased.
+    max_variants : int
+        The maximum number of variants (rows) to display. If there are
+        more variants than this then the table is truncated.
+    max_samples : int
+        The maximum number of samples (columns) to display. If there are
+        more samples than this then the table is truncated.
+
+    Returns
+    -------
+    GenotypeDisplay
+        A printable object to display genotype information.
+    """
+
+    variants = ds["variant_id"] if "variant_id" in ds else ds["variant_position"]
+
+    gt = _truncate(
+        ds["call_genotype"], variants, ds["sample_id"], max_variants, max_samples
+    ).astype(str)
+    missing = _truncate(
+        ds["call_genotype_mask"], variants, ds["sample_id"], max_variants, max_samples
+    )
+    calls = xr.where(missing, ".", gt)  # type: ignore[no-untyped-call]
+
+    def make_unphased_genotype(x: ArrayLike) -> str:
```

Comment:
Is it possible for this to be anything other than a DataArray type?

Reply:
This code no longer exists in the latest commit.